### PR TITLE
Adding support for KSZ9477 managed ethernet switch for saluki-v3

### PR DESCRIPTION
**Pull request by jlaitine**
## Summary

Adding code for saluki-v3 ethernet switch configuration.

Tested to be working on saluki-v3

### CODE REVIEW

File: drivers/net/Make.defs [hunk 1]
```diff
@@ -80,6 +80,15 @@ ifeq ($(CONFIG_ARCH_PHY_INTERRUPT),y)
   CSRCS += phy_notify.c
 endif
 
+ifeq ($(CONFIG_NET_KSZ9477),y)
+  CSRCS += ksz9477.c
+
+ifeq ($(CONFIG_NET_KSZ9477_I2C),y)
+CSRCS += ksz9477_i2c.c
```
Comment (jlaitine) [inline drivers/net/Make.defs:87]: indent

File: drivers/net/Kconfig [hunk 2]
```diff
@@ -392,6 +392,30 @@ menuconfig NET_W5500
 		References:
 		W5500 Datasheet, Version 1.0.9, 2013 WIZnet Co., Ltd.
 
+config NET_KSZ9477
+	bool "Management interface for ksz9477 ethernet switch"
+	default n
+	---help---
+		Support for Microchip/Micrel KSZ9477 manaaged switch. To use this,
```
Comment (jlaitine) [inline drivers/net/Kconfig:399]: typo. manaaged

File: drivers/net/ksz9477_i2c.c [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                     read_msg->len - sizeof(read_msg->reg), true);
+
+  return I2C_TRANSFER(ksz9477_i2c_bus, msg, 2);
+}
+
+int ksz9477_write(struct ksz9477_transfer_s *write_msg)
+{
+  struct i2c_msg_s msg;
+
+  /* Set up to write the address */
```
Comment (jlaitine) [inline drivers/net/ksz9477_i2c.c:85]: address + data

File: drivers/net/ksz9477_i2c.c [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                     write_msg->len, false);
+
+  return I2C_TRANSFER(ksz9477_i2c_bus, &msg, 1);
+}
+
+/****************************************************************************
+ * Name: ksz9477_i2c_init
+ *
+ * Description:
+ *   Switches the ksz9477's SGMII port into PHY mode and sets the
```
Comment (jlaitine) [inline drivers/net/ksz9477_i2c.c:97]: Update description, this was copy paste

File: drivers/net/ksz9477_i2c.c [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <debug.h>
```
Comment (pussuw) [inline drivers/net/ksz9477_i2c.c:26]: I don't think this header is needed here?

File: drivers/net/ksz9477_reg.h [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+/* Global registers */
+
+#define KSZ9477_ID0                        0x0
+#define KSZ9477_ID1                        0x1
+#define KSZ9477_ID2                        0x2
+#define KSZ9477_ID3                        0x3
+
+/* Port registers */
+
+#define KSZ9477_PORT_REG(port,x)           ((((uint16_t)port) << 12) | (x))
```
Comment (pussuw) [inline drivers/net/ksz9477_reg.h:50]: (port) ?

**Review by pussuw — APPROVED** [review]
LGTM, the comments are irrelevant, fix if you wish

File: drivers/net/ksz9477_i2c.c (hunk 5 above)
Comment (jlaitine) [reply drivers/net/ksz9477_i2c.c:26]: cool, I'll remove it, thanks

File: drivers/net/ksz9477_reg.h (hunk 6 above)
Comment (jlaitine) [reply drivers/net/ksz9477_reg.h:50]: Yes, definitely. I wil also change x -> reg for clarity

**Review by pussuw — APPROVED** [review]

**Comment by jlaitine** [comment]
I'll try upstreaming before merging this in

**Comment by jlaitine** [comment]
This change is now upstreamed, and this is direct cherry-pick from upstream. Merging->

**Pull request merged**